lib/main.js

```diff
@@ -0,0 +1,110 @@
+'use babel'
+
+import Path from 'path'
+import {CompositeDisposable} from 'atom'
+import {spawnWorker} from './helpers'
+import escapeHTML from 'escape-html'
+
+export default {
+  config: {
+    lintHtmlFiles: {
+      title: 'Lint HTML Files',
+      description: 'You should also add `eslint-plugin-html` to your .eslintrc plugins',
+      type: 'boolean',
+      default: false
+    },
+    useGlobalEslint: {
+      title: 'Use global ESLint installation',
+      description: 'Make sure you have it in your $PATH',
+      type: 'boolean',
+      default: false
+    },
+    showRuleIdInMessage: {
+      title: 'Show Rule ID in Messages',
+      type: 'boolean',
+      default: true
+    },
+    disableWhenNoEslintrcFileInPath: {
+      title: 'Disable when no .eslintrc is found',
+      type: 'boolean',
+      default: true
+    }
+  },
+  activate: function() {
+    require('atom-package-deps').install('linter-eslint')
+
+    this.subscriptions = new CompositeDisposable()
+    this.active = true
+    this.worker = null
+    this.scopes = ['source.js', 'source.jsx', 'source.js.jsx', 'source.babel', 'source.js-semantic']
+
+    const embeddedScope = 'source.js.embedded.html'
+    this.subscriptions.add(atom.config.observe('linter-eslint.lintHtmlFiles', lintHtmlFiles => {
+      if (lintHtmlFiles) {
+        this.scopes.push(embeddedScope)
+      } else {
+        if (this.scopes.indexOf(embeddedScope) !== -1) {
+          this.scopes.splice(this.scopes.indexOf(embeddedScope), 1)
+        }
+      }
+    }))
+
+    // Reason: I (steelbrain) have observed that if we spawn a process while atom is starting up, it can increase startup time by several seconds,
```

Comment:
Mind splitting this across lines a bit earlier? If I have to scroll to see it on GitHub it's too long :stuck_out_tongue:.